rust/connlib/clients/apple/src/lib.rs

```diff
@@ -104,9 +104,9 @@ impl Callbacks for CallbackHandler {
         Ok(())
     }
 
-    fn on_add_route(&self, route: IpNetwork) -> Result<(), Self::Error> {
+    fn on_add_route(&self, route: IpNetwork) -> Result<RawFd, Self::Error> {
         self.inner.on_add_route(route.to_string());
-        Ok(())
+        Ok(-1)
```

Comment:
If this is on purpose, we should leave a comment here explaining why we return `-1` here.

Reply:
Changed to use `Option` and added comment on the trait.